packages/react-router/modules/Router.js

```diff
@@ -53,14 +54,18 @@ class Router extends React.Component {
   render() {
     return (
       <RouterContext.Provider
-        children={this.props.children || null}
         value={{
           history: this.props.history,
           location: this.state.location,
           match: Router.computeRootMatch(this.state.location.pathname),
           staticContext: this.props.staticContext
         }}
-      />
+      >
+        <HistoryContext.Provider
+          children={this.props.children || null}
```

Comment:
ask question! why pass the 'children' to the provider? who can accept it, and use it! 
I have a try, but haven't got 'children' but 'value'

Reply:
We just pass the `children` prop down, because the `children` prop of `<Router />` are wrapped by the history and the router context.

Reply:
That's the same syntax as:

```jsx
<HistoryContext.Provider value={this.props.history}>
  {this.props.children || null}
</HistoryContext.Provider>
```
as in normal React context API. I just followed the old code style without changing it.

Reply:
@camillalo `children` is a special prop in React that can be written in two way, as @illuminist just told.